runner.py

```diff
@@ -0,0 +1,134 @@
+import runner_run_funcs
+
+SIM_TYPE_FED_LEARNING = "fed_learning"
+SIM_TYPE_FED_AVG_WITH_DP = "fed_avg_with_dp"
+
+DATA_GEN_TYPE_DATA_FROM_FILE = "file_data"
+DATA_GEN_TYPE_BLOB = "data_gen_blob"
+DATA_GEN_TYPE_RAND = "data_gen_rand"
+
+P_KEY_DATA_FILE_PATH = "data_file_path"
+
+P_KEY_NUM_SAMPLES = "num_samples"
+P_KEY_NUM_LABELS = "num_labels"
+P_KEY_NUM_FEATURES = "num_features"
+P_KEY_NUM_USERS = "num_users"
+
+P_KEY_NUM_ROUNDS = "num_rounds"
+P_KEY_BATCH_SIZE = "batch_size"
+P_KEY_NUM_EPOCHS = "num_epochs"
+
+
+class RunFuncAndReqParams:
```

Comment:
I think we can safely drop this class - it doesn't 'carry it's own weight'.   A code smell for a class that can be removed is that you've got internal state, but no methods attached to the class.

I would replace this with either a simple 2-tuple, or possibly a dict.

Reply:
Yeah I agree. A tuple might make more sense for two fields when it's self contained within a module.